Include/pyport.h

```diff
@@ -768,13 +768,71 @@ extern char * _getpty(int *, int, mode_t, int);
  */
 
 #ifdef WORDS_BIGENDIAN
-#define PY_BIG_ENDIAN 1
-#define PY_LITTLE_ENDIAN 0
+#  define PY_BIG_ENDIAN 1
+#  define PY_LITTLE_ENDIAN 0
 #else
-#define PY_BIG_ENDIAN 0
-#define PY_LITTLE_ENDIAN 1
+#  define PY_BIG_ENDIAN 0
+#  define PY_LITTLE_ENDIAN 1
 #endif
 
+#if defined(__clang__) || \
+    (defined(__GNUC__) && \
+     ((__GNUC__ >= 5) || (__GNUC__ == 4) && (__GNUC_MINOR__ >= 8)))
+   /* __builtin_bswap32() available in clang and GCC 4.8 */
+#  define _PY_HAVE_BUILTIN_BSWAP
+#endif
+
+static inline uint16_t
+_Py_bswap16(uint16_t word)
+{
+#ifdef _PY_HAVE_BUILTIN_BSWAP
+    return __builtin_bswap16(word);
+#else
+    return (((word >> 8) & 0x00FF)
+            | ((word << 8) & 0xFF00));
+#endif
+}
+
+static inline uint32_t
+_Py_bswap32(uint32_t word)
+{
+#ifdef _PY_HAVE_BUILTIN_BSWAP
+    return __builtin_bswap32(word);
+#else
+    word = (   (word & 0xFF00FF00UL) >> 8)
```

Comment:
I am not sure, but it may be a benefit from using the same constant: `(word >> 8) & 0x00FF00FFUL`. It may also be a benefit from using `&` and `>>` in different orders. I did not tested this.

Reply:
I let you test it. I don't really care of the performance on this code path, since efficient builtin functions are used on most platforms.